2/contrib/s2i/run

```diff
@@ -186,12 +186,21 @@ if [ ! -e ${JENKINS_HOME}/configured ]; then
     touch ${JENKINS_HOME}/configured
 else  
   if [ ! -z "${OVERRIDE_PV_CONFIG_WITH_IMAGE_CONFIG}" ]; then
-    echo "Overriding jenkins config.xml stored in ${JENKINS_HOME}/config.xml"
-    rm -f ${JENKINS_HOME}/config.xml
-	
-    create_jenkins_config_xml
+    echo "Overriding jenkins configuration stored in ${JENKINS_HOME}"
+
+    # When overriding config, we should override every config file except plugins which might be a special case (?)
+    for f in ${image_config_dir}/*; do
+      fileName=${f#${image_config_dir}'/'}
+      if [ -e "${JENKINS_HOME}/${fileName}" ]; then
+        echo "---> Removing ${JENKINS_HOME}/${fileName}..."
+        rm -rf "${JENKINS_HOME}/${fileName}"
+      fi
+    done
```

Comment:
is this actually necessary?  seems like the cp is just going to overwrite anyway.

Reply:
I hesitated to put this.
The cp will overrite existing files but it will not remove existing content in the directory.
Let say for example if we want to rename or delete a job from our git repository, this job will actually remains as cp command will not remove it.

Reply:
hm.  but this means if your new config contains even a single job, all existing jobs will be deleted (because the jobs dir will be removed).  Which I guess is what you asked for by asking for the override behavior, but it's a bit scary.

Reply:
Indeed! 
That’s why I hesitated but I decided to do it that way for these reasons: 
1. Regarding the doc and the override aspect, this seems logical to me.
1. If you look the assemble script, it does the same thing for the _Openshift example_, it deletes the job if other jobs are provided (but indeed it’s only in the image layer here)
1. IMO, once you start configuring jenkins in your image, you should do it for every single configuration to manage your infrastructure as code.

To summarize my opinion, the persistent volume makes sense for the workspace, to keep some build history or for stash/caching but not to store the configuration. It also emphases the immutability of the docker image.

At the end, I’m open to any suggestion, so if you have a better one, please shoot, I would gladly adapt it. 

My goal is to allow user to manage the jenkins configuration in the image, so as long we can do that, I would be happy! :-)

Reply:
yeah i think your justification is reasonable.